File: product-services/src/main/java/com/microservices/productservices/service/TagService.java
package com.microservices.productservices.service;

import com.microservices.productservices.payload.request.TagRequest;
import com.microservices.productservices.payload.response.TagResponse;

import java.util.List;
import java.util.UUID;

public interface TagService {

    TagResponse create(TagRequest tagRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    void isDisplay(UUID id);

    TagResponse getById(UUID id);

    List<TagResponse> getAll();

    TagResponse update(UUID id, TagRequest tagRequest);

    TagResponse delete(UUID id);
}

File: reactjs-app/src/services/TagService.js
import httpAxios from '../httpAxios';

const TagService = {
    create: (brandData) => {
        return httpAxios.post(`product-services/api/tags/create`, brandData);
    },
    setImage: (data) => {
        return httpAxios.put(`product-services/api/tags/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`product-services/api/tags/get-by-id/${id}`);
    },
    getByUser: (userId) => {
        return httpAxios.get(`product-services/api/tags/get-by-user/${userId}`);
    },
    getAll: () => {
        return httpAxios.get(`product-services/api/tags/get-all`);
    },
    update: (id, brand) => { 
        return httpAxios.put(`product-services/api/tags/update/${id}`, brand);
    },
    sitchStatus: (id) => {
        return httpAxios.put(`product-services/api/tags/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`product-services/api/tags/trash/${id}`);
    },
    display: (id) => {
        return httpAxios.put(`product-services/api/tags/display/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`product-services/api/tags/delete/${id}`);
    },
}
export default TagService;
